osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderRepeat.cs

```diff
@@ -87,21 +85,15 @@ protected override void OnApply()
 
         protected override void UpdateInitialTransforms()
         {
-            // When snaking in is enabled, the first end circle needs to be delayed until the snaking completes.
-            bool delayFadeIn = DrawableSlider.SliderBody?.SnakingIn.Value == true && HitObject.RepeatIndex == 0;
-
-            animDuration = Math.Min(300, HitObject.SpanDuration);
-
-            this
-                .FadeOut()
-                .Delay(delayFadeIn ? (Slider?.TimePreempt ?? 0) / 3 : 0)
-                .FadeIn(HitObject.RepeatIndex == 0 ? HitObject.TimeFadeIn : animDuration);
+            ApplyRepeatFadeIn(this);
```

Comment:
Again, just going off of the videos you provided, I don't think this can't be applied to `this`. I think it must be applied to `CirclePiece` *specifically*.

In your video [from the OP](https://github.com/ppy/osu/pull/32638#issue-2959784799), showing stable:

![Screenshot 2025-04-03 at 09 08 25](https://github.com/user-attachments/assets/8abbca15-5218-465a-9267-dbc4d1bc238a)

Note how the repeat arrow is *significantly more opaque* than the circle piece.

On your [latest video](https://github.com/ppy/osu/pull/32638#issuecomment-2774657993), however:

![Screenshot 2025-04-03 at 09 09 20](https://github.com/user-attachments/assets/2730aec8-636d-4f3e-9a7d-b579be43e3ae)

Reply:
Yeah it didn't seem to match in the video after the change. I tried initially applying to `CirclePiece` but things exploded. I'll revisit.

Reply:
Okay I think I get it, and I think it visually matches now..


https://github.com/user-attachments/assets/06644891-9129-4c2b-b851-34dea9c7d401